### DISCUSSION TITLE
python papermerge API client

### DISCUSSION THREAD

**Discussion by p-rintz** [ciur/papermerge#483] [Show and tell]
Hi @ciur ,

I wanted to show a rather early version of a python papermerge API client I'm working on, [pypapermerge](https://github.com/p-rintz/pypapermerge).

Before I continue with writing it I wanted to show it to you, in case you're already working on something like it privately.
I also just wanted to hear your thoughts about this, generally.

Thanks!

https://github.com/p-rintz/pypapermerge

**Comment by ciur** [comment]
@p-rintz I am not sure if this is right approach.
Definitely there should be a REST API client, however writing it by hand may not be the right approach.

As far as I know, the common approach today is to use generate clients (e.g. python client) from openapi schema yaml file.
The openapi schema yaml file in turn is generated from (from REST API backend source code).
I know only the theory though, I didn't played with generating REST API client yet and I don't know if my argument is valid - just for you to keep in mind; if you consider your way is the right one - just ignore my blah here!


PS:
[This](https://openapi-generator.tech/) may be the starting point (I just googled "automatically generated REST API clients python from openapi) :stuck_out_tongue: .

**Comment by p-rintz** [comment]
Heya,

yeah, I did know of them, and before I started it I did see the generators for it, but, honestly, when I tried generating the client and then looked at the code, it looked not very legible. 

So I decided I might as well write one to learn some stuff about pytest and mkdocs :laughing:
